components/uniqueness/backing-store-impl/src/integrationTest/kotlin/net/corda/uniqueness/backingstore/impl/JPABackingStoreImplIntegrationTests.kt

```diff
@@ -0,0 +1,693 @@
+import net.corda.uniqueness.datamodel.impl.UniquenessCheckErrorInputStateConflictImpl
+import net.corda.uniqueness.datamodel.impl.UniquenessCheckErrorReferenceStateConflictImpl
+import net.corda.uniqueness.datamodel.internal.UniquenessCheckRequestInternal
+import net.corda.uniqueness.utils.UniquenessAssertions
+import net.corda.uniqueness.utils.UniquenessAssertions.toErrorType
+import net.corda.v5.application.uniqueness.model.UniquenessCheckResult
+import net.corda.v5.application.uniqueness.model.UniquenessCheckErrorInputStateUnknown
+import net.corda.v5.application.uniqueness.model.UniquenessCheckErrorMalformedRequest
+import net.corda.v5.application.uniqueness.model.UniquenessCheckErrorTimeWindowOutOfBounds
+import net.corda.v5.application.uniqueness.model.UniquenessCheckErrorReferenceStateUnknown
+import net.corda.v5.application.uniqueness.model.UniquenessCheckErrorInputStateConflict
+import net.corda.v5.application.uniqueness.model.UniquenessCheckErrorReferenceStateConflict
+import net.corda.v5.application.uniqueness.model.UniquenessCheckStateRef
+import net.corda.v5.crypto.SecureHash
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.TestInstance
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.Nested
+import org.junit.jupiter.api.Disabled
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.assertThrows
+import org.junit.jupiter.api.assertDoesNotThrow
+import org.junit.jupiter.api.assertAll
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.ValueSource
+import org.mockito.Mockito
+import org.mockito.kotlin.whenever
+import org.mockito.kotlin.doReturn
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.any
+import org.mockito.kotlin.times
+import org.mockito.kotlin.never
+import java.time.Instant
+import java.time.LocalDateTime
+import java.time.LocalDate
+import java.time.ZoneOffset
+import java.time.Clock
+import javax.persistence.EntityManagerFactory
+import javax.persistence.EntityExistsException
+import javax.persistence.RollbackException
+import javax.persistence.OptimisticLockException
+import javax.persistence.PersistenceException
+import javax.persistence.QueryTimeoutException
+import kotlin.reflect.full.createInstance
+import kotlin.test.assertEquals
+
+/**
+ * Hint: To run tests against PostgreSQL, follow the steps in the link below.
+ * https://github.com/corda/corda-runtime-os/wiki/Debugging-integration-tests#debugging-integration-tests-with-postgres
+ */
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class JPABackingStoreImplIntegrationTests {
+    private lateinit var backingStoreImpl: JPABackingStoreImpl
+    private lateinit var lifecycleCoordinator: LifecycleCoordinator
+    private lateinit var lifecycleCoordinatorFactory: LifecycleCoordinatorFactory
+
+    companion object {
+        private val UPPER_BOUND = LocalDate.of(2200, 1, 1).atStartOfDay().toInstant(ZoneOffset.UTC)
```

Comment:
```suggestion
        private val DEFAULT_TIME_WINDOW_UPPER_BOUND = LocalDate.of(2200, 1, 1).atStartOfDay().toInstant(ZoneOffset.UTC)
```

Bit clearer in my opinion

Reply:
Updated

Reply:
Updated